### ISSUE TITLE
How to Pass HttpClient from ASP.NET Core HttpClientFactory to Stripe?

### ISSUE THREAD

**Issue by EdBlankenship** [stripe/stripe-dotnet#1486]
Do you have an example of passing an `HttpClient `from an `HttpClientFactory `that's available for optimizing `HttpClient` usage in .NET Core 2.1 and forward?

As an example, [Twilio allows you to provide a custom `HttpClient`](https://www.twilio.com/docs/libraries/csharp-dotnet/custom-http-clients-dot-net-core) like below.

```cs
var httpClient = httpClientFactory.CreateClient("TwilioClient");
var twilioHttpClient = new SystemNetHttpClient(httpClient);
            
_twilioRestClient = new TwilioRestClient(options.Value.AccountSid,
     options.Value.AuthToken,
     httpClient: twilioHttpClient
     );
```

More info about `HttpClientFactory` that might be useful:
- [Use HttpClientFactory to Implement Resilient HTTP Requests](https://docs.microsoft.com/en-us/dotnet/standard/microservices-architecture/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests) 
- [You're Using HttpClient Wrong and It Is Destabilizing Your Software](https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/)

**Comment by ob-stripe** [comment]
Hi @EdBlankenship. Currently there is no way to use a custom `HttpClient`. Note that the second article you linked doesn't apply to Stripe.net: we do use a single static instance of `HttpClient`, so connections are persisted.

Nevertheless, we'll look into passing a custom `HttpClient` to the library in a future update. Tagging this as `future` for now.

**Comment by EdBlankenship** [comment]
Thanks @ob-stripe for taking a look into it and considering it.

**Comment by ob-stripe** [comment]
Hi! Sorry it took so long, but this is fixed as of version 27.0.0. You can now pass custom `System.Net.Http.HttpClient` instances to the library.  You can find a few examples in this wiki page: https://github.com/stripe/stripe-dotnet/wiki/Advanced-client-usage.
